workflow/controller/operator.go

```diff
@@ -1680,6 +1680,11 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 			woc.addChildNode(retryNodeName, nodeName)
 			node = nil
 
+			// It has to be one child at least
+			if lastChildNode != nil {
+				RetryOnDifferentHost(retryNodeName)(*woc.retryStrategy(processedTmpl), woc.wf.Status.Nodes, processedTmpl)
```

Comment:
The retry node has the same `Name` and `ID` so can we stick with `retryNodeName` or I should change and use `ID` here?